android/build/common.xml

```diff
@@ -378,22 +382,68 @@ Common ant tasks and macros for building Android-based Titanium modules and proj
 		</sequential>
 	</macrodef>
 
+	<macrodef name="gperf">
+		<attribute name="base.directory" default="${basedir}"/>
+		<attribute name="in.file"/>
+		<attribute name="out.file"/>
+		<sequential>
+			<exec executable="${gperf.tool}" dir="@{base.directory}" output="@{out.file}" failonerror="true">
+				<arg value="-L"/>
+				<arg value="C++"/>
+				<arg value="-E"/>
+				<arg value="-t"/>
+				<!-- Set to "value", not "file", because Win32 "gperf.exe" won't escape '\' in path. -->
+				<arg value="@{in.file}"/>
+			</exec>
+		</sequential>
+	</macrodef>
+
 	<target name="generate.kroll.v8.bindings" depends="build.kroll.apt">
 		<generate.bindings dest="${kroll.v8.project.dir}/generated"/>
 	</target>
 
 	<target name="build.kroll.v8.so" depends="generate.kroll.v8.bindings,copy.kroll.v8.jars">
-		<condition property="ndk.build.args" value="" else="-j8">
-			<os family="windows"/>
-		</condition>
 		<property name="kroll.v8.build.x86" value=""/>
+
+		<!-- Generate a "KrollNativeBindings.h" file with perfect hashes via gperf tool. -->
+		<!-- Note: "in.file" string is inserted into file. Use relative path since absolute may contain user name. -->
+		<gperf
+			base.directory="${kroll.v8.project.dir}"
```

Comment:
Is this meant to be on the same line as
```XML
<gperf
```
?

Reply:
I purposely put the attribute on the next line, lining it up with the rest of the wrapped attributes. We do this in other places in the xml file as well. Admittedly, it looks better with 4 spaced tabs.